library/ajax/track_events.php

```diff
@@ -0,0 +1,136 @@
+    //$result = reportUsageData(); // todo: remove this line. for testing until background task is implemented.
+} elseif ($data['action'] === 'reportUsageData') {
+    $result = reportUsageData();
+    echo json_encode($result);
+} else {
+    http_response_code(400);
+    echo json_encode(array("error" => "Invalid action"));
+}
+
+function reportClicks()
+{
+    global $data;
+    $eventType = $data['eventType'] ?? '';
+    $eventLabel = $data['eventLabel'] ?? '';
+    $eventUrl = $data['eventUrl'] ?? '';
+    $eventTarget = $data['eventTarget'] ?? '';
+    $currentTime = date("Y-m-d H:i:s");
+
+    if (empty($eventType) || empty($eventLabel)) {
+        http_response_code(400);
+        echo json_encode(array("error" => "Missing required fields"));
+        exit;
+    }
+
+    $sql = "INSERT INTO track_events (event_type, event_label, event_url, event_target, first_event, last_event, count)
+        VALUES (?, ?, ?, ?, ?, ?, ?)
+        ON DUPLICATE KEY UPDATE 
+          event_url    = ?,
+          event_target = ?,
+          last_event   = ?,
+          count        = count + 1";
+
+    $params = array(
+        $eventType,
+        $eventLabel,
+        $eventUrl,
+        $eventTarget,
+        $currentTime,
+        $currentTime,
+        1,
+        // Update values:
+        $eventUrl,
+        $eventTarget,
+        $currentTime
+    );
+    $result = sqlStatement($sql, $params);
+
+    if ($result) {
+        echo json_encode(array("success" => true));
+    } else {
+        http_response_code(500);
+        echo json_encode(array("error" => "Database insertion/update failed"));
+    }
+}
+
+// TODO needs work
+// TODO call from a background task.
+function reportUsageData()
+{
+    global $data;
+
+    $site_uuid = UniqueInstallationUuid::getUniqueInstallationUuid() ?? '';
+    if (empty($site_uuid)) {
+        error_log("Site UUID not found.");
+        return false;
+    }
+
+    $versionService = new VersionService();
+    $endpoint = "https://reg.open-emr.org/api/usage?SiteID=" . urlencode($site_uuid);
+    $interval = date("Ym", strtotime("-30 Days"));
+    $time_zone = sqlQuery("SELECT `gl_value` as zone FROM `globals` WHERE `gl_name` = 'gbl_time_zone' LIMIT 1")['zone'] ?? null;
+    // Query the track_events table for usage data.
+    $sql = "SELECT event_type, event_label, event_url, event_target, first_event, last_event, count FROM track_events";
+    $result = sqlStatement($sql);
+    $usageRecords = array();
+    while ($row = sqlFetchArray($result)) {
+        $usageRecords[] = $row;
+    }
+    // Collect instance data.
+    $localeData = array(
+        'site_uuid' => $site_uuid,
+        'reporting_interval' => $interval,
+        'location' => '',
+        'time_zone' => $time_zone ?? $GLOBALS['gbl_time_zone'] ?? '',
+        'locale' => locale_get_default(),
+        'version' => $versionService->asString(),
+        'distribution' => getenv('OPENEMR_DOCKER_ENV_TAG', true) ?? $versionService->asString(),
```

Comment:
prob just set above 'distribution' to blank if it doesn't exist

Also, would it make sense to record OS info (maybe in os column or something like that) from php_uname() command?